capsules/extra/src/process_info_driver.rs

```diff
@@ -0,0 +1,254 @@
+// Licensed under the Apache License, Version 2.0 or the MIT License.
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+// Copyright Tock Contributors 2025.
+
+//! Allow userspace to inspect and control processes on the board.
+//!
+//! ## Warning!
+//!
+//! This capsule is designed for testing and experimental use cases only. It
+//! should not be used in production! It was originally written for use in an
+//! educational tutorial to make it easy to interact with processes stored on
+//! the board from userspace using a screen.
+//!
+//! This capsule does require a capability to also indicate that this interacts
+//! with processes in a way that common capsules should not.
+//!
+//! ## Commands
+//!
+//! - 0: Check driver exists.
+//! - 1: Get the count of processes running on the board.
+//! - 2: Fill the allow RW buffer with the process IDs for the running processes
+//!   on the board. Returns the number of running processes.
+//! - 3: Fill the allow RW buffer with the short IDs for the running processes
+//!   on the board. Returns the number of running processes.
+//! - 4: Put the name of the process specified by the process ID in `data1` in
+//!   the allow RW buffer.
```

Comment:
Should this maybe not also return the length, such that userspace can doesn't print uninitialized data and can re-try this operation with a larger buffer if the name exceeds the allow space?

Reply:
Yes. I also made the impl more robust with better error handling.